src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -493,21 +493,14 @@ impl Device for VfioDevice {
 
                 if self.bus_mode == VfioBusMode::PCI {
                     for hostdev in self.devices.iter_mut() {
-                        if hostdev.guest_pci_path.is_none() {
-                            // guest_pci_path may be empty for certain hypervisors such as
-                            // dragonball
-                            hostdev.guest_pci_path = Some(
-                                generate_guest_pci_path(hostdev.bus_slot_func.clone())
-                                    .map_err(|e| anyhow!("generate pci path failed: {:?}", e))?,
-                            );
-                        }
-
-                        // Safe to call unwrap here because of previous assignment.
-                        let pci_path = hostdev.guest_pci_path.clone().unwrap();
+                        hostdev.guest_pci_path =
+                            generate_guest_pci_path(hostdev.bus_slot_func.clone())
```

Comment:
This would break for cloud-hypervisor and other hypervisors. We assign the `guest_pci_path` in CH when the device is attached:
https://github.com/kata-containers/kata-containers/blob/04de6eb4fd9bbd5440d8a2f48e26806ba3756eb9/src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs#L220

The same should be done for qemu and others. 
The guest_pci_path is empty only for dragonball, since it doesnt use PCIe at the moment. 
So overwriting this would break for CH and others. I had added back the above conidtional assignment for drangonball based on your suggestion in my other PR.

Reply:
If the assignment to guest_pci_path is necessary for dragonball, my suggestion would be to move this to dragonball specific `inner_device.rs` similar to how it is being done currently for CH.

Reply:
fixed